packages/web-app-files/tests/components/BatchActions.spec.js

```diff
@@ -0,0 +1,250 @@
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import DesignSystem from 'owncloud-design-system'
+import Vuex from 'vuex'
+import getters from '../../src/store/getters'
+import stubs from '../../../../tests/unit/stubs'
+import BatchActions from '../../src/components/BatchActions.vue'
+
+const localVue = createLocalVue()
+localVue.use(DesignSystem)
+localVue.use(Vuex)
+
+const selectedFiles = ['lorem.txt', 'owncloud.png']
+const componentStubs = { ...stubs, translate: true }
+const elSelector = {
+  copyButton: '#copy-selected-btn',
+  moveButton: '#move-selected-btn',
+  deleteButton: '#delete-selected-btn',
+  restoreButton: '#restore-selected-btn'
+}
+
+describe('Batch Actions component', () => {
+  const mountOptions = {
+    localVue
+  }
+
+  describe('files page', () => {
+    const $route = {
+      name: 'files-personal'
+    }
+
+    describe('when no items are selected for batch action', () => {
+      const options = {
+        ...mountOptions,
+        mocks: {
+          $route: {
+            ...$route,
+            meta: {
+              hasBulkActions: false
+            }
+          }
+        }
+      }
+
+      it('should not display action buttons', () => {
+        const wrapper = shallowMount(BatchActions, options)
+        const actionButtons = wrapper.findAll('oc-grid-stub oc-button-stub')
+
+        expect(actionButtons.length).toEqual(0)
+        expect(wrapper.exists()).toBeTruthy()
+      })
+    })
+
+    describe('when items are selected for batch action', () => {
+      const store = createStore({ selected: selectedFiles })
+
+      const options = {
+        ...mountOptions,
+        stubs: { ...componentStubs, 'oc-button': false },
+        store,
+        mocks: {
+          $route: {
+            ...$route,
+            meta: {
+              hasBulkActions: true
+            }
+          }
+        }
+      }
+
+      let wrapper
+      let copyButton
+      let moveButton
+      let deleteButton
+
+      BatchActions.computed.canMove = jest.fn(() => true)
+      const spyTriggerLocationPicker = jest
+        .spyOn(BatchActions.methods, 'triggerLocationPicker')
+        .mockImplementation()
+      const spyDeleteResourcesDisplayDialog = jest
+        .spyOn(BatchActions.mixins[1].methods, '$_deleteResources_displayDialog')
+        .mockImplementation()
```

Comment:
Should we mock component methods like above?
Because mocking as below:
```js
wrapper.vm.myFunc = jest.fn()
wrapper.vm.$forceUpdate()
```
would be same as using `setMethods`.
see https://vue-test-utils.vuejs.org/api/wrapper/#setmethods

Reply:
I think that is fine. @kulmann thoughts?